messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -88,6 +85,10 @@
   public static String clusterNamePrefix = "databusCluster";
   private Boolean readFromLocalStream;
   private int numList = 0;
+  private  List<String> collectorsWithReaders = new ArrayList<String>();
```

Comment:
I feel we should not make createPartitionReaders() method as synchronized. If consumer is invoked next() (i.e. syncronized)  and blocked on queue for getting message then there is no way to be executed createPartitionReader() call as both craetePartitionReader and next() are synchronized methods.